Tools/clinic/clinic.py

```diff
@@ -319,51 +275,32 @@ def linear_format(s: str, **kwargs: str) -> str:
         if not value:
             continue
 
-        value = textwrap.indent(rstrip_lines(value), indent)
-        add(value)
-        add('\n')
+        stripped = [line.rstrip() for line in value.split("\n")]
+        value = textwrap.indent("\n".join(stripped), indent)
+        lines.extend([value, "\n"])
 
-    return output()[:-1]
+    return "".join(lines[:-1])
 
-def indent_all_lines(s: str, prefix: str) -> str:
+def _add_prefix_and_suffix(s: str, prefix: str = "", suffix: str = "") -> str:
     """
-    Returns 's', with 'prefix' prepended to all lines.
+    Return 's', with 'prefix' prepended and 'suffix' appended to all lines.
 
-    If the last line is empty, prefix is not prepended
-    to it.  (If s is blank, returns s unchanged.)
+    If the last line is empty, it remains unchanged.
+    If s is blank, returns s unchanged.
 
     (textwrap.indent only adds to non-blank lines.)
     """
-    split = s.split('\n')
-    last = split.pop()
-    final = []
-    for line in split:
-        final.append(prefix)
-        final.append(line)
-        final.append('\n')
+    *split, last = s.split("\n")
+    lines = [prefix + line + suffix + "\n" for line in split]
```

Comment:
Optional, just my personal preference:

```suggestion
    lines = [f"{prefix}{line}{suffix}\n" for line in split]
```

Reply:
I considered this, and I anticipated your suggestion ;) I think I find the + variant slightly more readable.

Reply:
Sure :)